test/gas-analytics/HubPool.RefundExecution.ts

```diff
@@ -0,0 +1,136 @@
+  await erc20.addMember(TokenRolesEnum.MINTER, owner.address);
+  return erc20;
+}
+
+// Construct tree with REFUND_CHAIN_COUNT leaves, each containing REFUND_TOKEN_COUNT sends
+async function constructSimpleTree() {
+  for (let i = 0; i < REFUND_CHAIN_COUNT; i++) {
+    _bundleLpFeeAmounts[i] = [];
+    _netSendAmounts[i] = [];
+    for (let j = 0; j < REFUND_TOKEN_COUNT; j++) {
+      _bundleLpFeeAmounts[i].push(LP_FEE);
+      _netSendAmounts[i].push(SEND_AMOUNT);
+
+      // Whitelist route
+      await hubPool.whitelistRoute(destinationChainIds[i], l1Tokens[i][j].address, randomAddress());
+    }
+
+    // Set adapter for destination chain ID:
+    await hubPool.setCrossChainContracts(destinationChainIds[i], mockAdapter.address, mockSpoke.address);
+  }
+  const leafs = buildPoolRebalanceLeafs(
+    destinationChainIds,
+    Object.values(l1Tokens),
+    _bundleLpFeeAmounts,
+    _netSendAmounts, // netSendAmounts.
+    _netSendAmounts // runningBalances.
+  );
+  const tree = await buildPoolRebalanceLeafTree(leafs);
+
+  return { leafs, tree };
+}
+
+describe("Gas Analytics: HubPool Relayer Refund Execution", function () {
+  before(async function () {
+    if (!process.env.GAS_TEST_ENABLED) this.skip();
+  });
+
+  beforeEach(async function () {
+    [owner, dataWorker, liquidityProvider] = await ethers.getSigners();
+    ({ hubPool, timer, weth, mockSpoke, mockAdapter } = await hubPoolFixture());
+
+    // Seed data worker with bond tokens.
+    await seedWallet(dataWorker, [], weth, consts.bondAmount.mul(10));
+    await weth.connect(dataWorker).approve(hubPool.address, consts.bondAmount.mul(10));
+
+    // Deploy test tokens for each chain ID
+    for (let i = 0; i < REFUND_CHAIN_COUNT; i++) {
+      destinationChainIds.push(i);
+      l1Tokens[i] = [];
+      for (let j = 0; j < REFUND_TOKEN_COUNT; j++) {
+        const _l1Token = await deployErc20(owner, `Test Token #${i}`, `T-${i}`);
+        l1Tokens[i].push(_l1Token);
+
+        // Mint data worker amount of tokens needed to bond a new root
+        await seedWallet(dataWorker, [_l1Token], undefined, consts.bondAmount);
+        await _l1Token.connect(dataWorker).approve(hubPool.address, consts.bondAmount);
+
+        // Mint LP amount of tokens needed to cover relay
+        await seedWallet(liquidityProvider, [_l1Token], undefined, STARTING_LP_AMOUNT);
+        await enableTokensForLP(owner, hubPool, weth, [_l1Token]);
+        await _l1Token.connect(liquidityProvider).approve(hubPool.address, STARTING_LP_AMOUNT);
+        await hubPool.connect(liquidityProvider).addLiquidity(_l1Token.address, STARTING_LP_AMOUNT);
+      }
+    }
+  });
+
+  it(`Tree with 1 Leaf: ${REFUND_SEND_COUNT} total transfers, ${REFUND_TOKEN_COUNT} different tokens`, async function () {
+    const { leafs, tree } = await constructSimpleTree();
+    const leafIndexToExecute = 0;
+
+    await hubPool.connect(dataWorker).initiateRelayerRefund(
```

Comment:
Should we be measuring the gas cost of this call as well?

Reply:
sure